sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java

```diff
@@ -283,4 +287,26 @@ public static String getApplicationId(ClientOptions clientOptions, HttpLogOption
             return null;
         }
     }
+
+    /**
+     * Creates {@link HttpHeaders} from the provided {@link ClientOptions}.
+     * <p>
+     * If {@code clientOptions} is null or {@link ClientOptions#getHeaders()} doesn't return any {@link Header} values
+     * null will be returned.
+     *
+     * @param clientOptions The {@link ClientOptions} used to create the {@link HttpHeaders}.
+     * @return {@link HttpHeaders} containing the {@link Header} values from {@link ClientOptions#getHeaders()} if
+     * {@code clientOptions} isn't null and contains {@link Header} values, otherwise null.
+     */
+    public static HttpHeaders createAddHeadersFromClientOptions(ClientOptions clientOptions) {
```

Comment:
`createHttpHeadersFromClientOptions` or just `createHttpHeaders(ClientOptions clientOptions)`.

Reply:
Whoops, I refactored this as it was returning `AddHeadersPolicy` before and I didn't like it as much, will fix the naming.